frontend/ui/screensaver.lua

```diff
@@ -170,17 +170,25 @@ function Screensaver:expandSpecial(message, fallback)
     if ui and ui.document then
         -- If we have a ReaderUI instance, use it.
         local doc = ui.document
-        currentpage = ui.view.state.page or currentpage
-        totalpages = doc:getPageCount() or totalpages
+        if doc:hasHiddenFlows() then
+            local currentpageAll = ui.view.state.page or currentpage
+            currentpage = doc:getPageNumberInFlow(ui.view.state.page or currentpageAll)
+            totalpages = doc:getTotalPagesInFlow(doc:getPageFlow(currentpageAll))
+            time_left_chapter = self:_calcAverageTimeForPages(ui.toc:getChapterPagesLeft(currentpageAll) or (totalpages - currentpage))
+            time_left_document = self:_calcAverageTimeForPages(totalpages - currentpage)
```

Comment:
What's wrong with `getTotalPagesLeft`, which looks like it has a much more elaborate logic for hidden flows?

Reply:
That's basically what I meant in https://github.com/koreader/koreader/pull/9905#discussion_r1045638788 btw

Reply:
Unless the intent was to show *current flow* info instead of *document* info?

In which case, I may have an issue with both

* that approach in itself
* the fact that it goes against the variable names to boot ;p.

Reply:
I agonized over that for a couple hours. I tried it with having a non-linear flow in the middle of the linear flow and got identical results with and without using it, so I'm not sure what situation that logic in `CreDocument:getTotalPagesLeft()` handles. _Seems_ like it's equivalent to `getTotalPagesInFlow()` minus `getPageNumberInFlow()` in both cases (both values which I already calculated right above it here so this should be slightly more performant in any case), so all that logic in the `else` block seems superfluous because getTotalPagesInFlow() should already be excluding any intervening non-linear flows, so all that manual work in the `else` clause is for... nothing?

Reply:
Regardless, I would follow ReaderFooter's example, if for no other reason that consistency ;).

And to shed more light on the matter, I'll ping our resident expert, the man with the flow, @Jellby ;).

Reply:
I'm curious about it. Still, even if we used `getTotalPagesLeft` for time left in book and its logic were different somehow, we'd still need new methods on Document to duplicate this logic to determine current page and total pages. And if we don't need those, then the current code already does the needful